engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorContextImpl.java

```diff
@@ -188,4 +188,87 @@ public Validator getValidator() {
 				traversableResolverResultCacheEnabled
 		);
 	}
+
+	/**
+	 * Context object storing all possible validator configurable properties.
+	 * There should be just one per {@link Validator} instance.
+	 */
+	static class ValidatorScopedContext {
```

Comment:
Maybe just `ValidatorContext`? And how about adding all the things scoped to the VF, too (or having another, dedicated context for that)? That way we'd get back to much leaner signatures of the methods passing through these objects.

Reply:
We already have `ValidatorContext` in the spec. That's why we tried to have another name.

@marko-bekhta my idea was to put this one in `ValidationContext` not `ValidatorContextImpl`. I don't think it has anything to do with `ValidatorContextImpl`.

Reply:
> And how about adding all the things scoped to the VF, too (or having another, dedicated context for that)? That way we'd get back to much leaner signatures of the methods passing through these objects.

I haven't thought about this yet. But, anyway, let's keep it for another PR.

Reply:
@gsmet that was my mistake - classes are together and I dropped the new one into a wrong one. 
@gunnarmorling +1 for updating the factory as well. Let's do it in another PR.